server/base/src/main/java/org/apache/accumulo/server/ServerInfo.java

```diff
@@ -175,10 +179,14 @@ public String getInstanceName() {
     return instanceName;
   }
 
-  public Credentials getCredentials() {
+  private Credentials loadCredentials() {
     return SystemCredentials.get(getInstanceID(), getSiteConfiguration());
   }
```

Comment:
This `loadCredentials` method should be inline'd into the constructor, and use the instanceId and siteConfig variables available in those constructors, to avoid calling instance methods on a half-constructed object. Since that change is trivial, I can go ahead and do it and merge this in.